src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -38,55 +38,46 @@ public SecretMasker() : this(default, default, default, default, default)
     {
     }
 
-    public SecretMasker(IEnumerable<RegexPattern>? regexSecrets,
+    public SecretMasker(IEnumerable<RegexPattern>? regexSecrets = null,
                         bool generateCorrelatingIds = false,
                         IRegexEngine? regexEngine = default,
                         string? defaultRegexRedactionToken = null,
                         string? defaultLiteralRedactionToken = null)
     {
-        _disposed = false;
-
-        RegexPatterns = regexSecrets != null
-            ? new HashSet<RegexPattern>(regexSecrets)
-            : new HashSet<RegexPattern>();
-
+        _regexPatterns = new HashSet<RegexPattern>(regexSecrets ?? []);
         _generateCorrelatingIds = generateCorrelatingIds;
 
-        ExplicitlyAddedSecretLiterals = new HashSet<SecretLiteral>();
-        EncodedSecretLiterals = new HashSet<SecretLiteral>();
-        LiteralEncoders = new HashSet<LiteralEncoder>();
+        _explicitlyAddedSecretLiterals = new HashSet<SecretLiteral>();
+        _encodedSecretLiterals = new HashSet<SecretLiteral>();
+        _literalEncoders = new HashSet<LiteralEncoder>();
 
         _regexEngine = regexEngine ??= CachedDotNetRegex.Instance;
 
         DefaultRegexRedactionToken = defaultRegexRedactionToken ?? RegexPattern.FallbackRedactionToken;
         DefaultLiteralRedactionToken = defaultLiteralRedactionToken ?? SecretLiteral.FallbackRedactionToken;
 
-        AddHighPerformancePatterns(RegexPatterns);
+        AddHighPerformancePatterns(_regexPatterns);
     }
 
-    // We don't permit secrets great than 5 characters in length to be
-    // skipped at masking time. The secrets that will be ignored when
-    // masking will N - 1 of this property value.
-    public static int MinimumSecretLengthCeiling { get; set; }
-
     [ThreadStatic]
     private static StringBuilder? s_stringBuilder;
 
-    public virtual string DefaultRegexRedactionToken { get; set; }
+    public string DefaultRegexRedactionToken { get; }
```

Comment:
>DefaultRegexRedactionToken [](http://example.com/codeflow?start=18&length=26)

These are no longer 'default' if they are instance data. Instead, they are simply the configured `RegexRedactionToken` etc.

Reply:
These were already instance data. Also, the regex redaction token is still a default. If c3ids are generated than it will prefer ID:c3id over this.

Reply:
OK, let's change these names in a future change.